.github/workflows/ci.yml

```diff
@@ -139,9 +142,43 @@ jobs:
           name: spectacular-package
           path: dist/packages/spectacular
           if-no-files-found: warn
+      - name: 'Spectacular: Upload docs build artifact'
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.spectacular-docs-package-name }}
+          path: dist/packages/spectacular-docs
+          if-no-files-found: warn
+
+  deploy-docs:
+    name: Deploy docs to GitHub Pages
+    # if: >
+    #   github.ref == 'refs/heads/main' && (github.event_name ==
+    #   'workflow_dispatch' || github.event_name == 'push')
+    needs: build
+    runs-on: ubuntu-latest
+
+    env:
+      package-path: ./spectacular-docs/
+
+    steps:
+      # necessary for JamesIves/github-pages-deploy-action
+      - uses: actions/checkout@v2
+
+      - run: echo ${{ github.ref }}
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: ${{ env.spectacular-docs-package-name }}
+          path: ${{ env.package-path }}
+
+      - name: Deploy docs to GitHub Pages
```

Comment:
Is this step updating the docs after every push using the same env?

In other words:

1. Are we creating a temp docs website per PR and just updating the final website after `main` is merge?
2. Are we always updating the same website env after every push?
3. Are we only updating the docs after the main branch merge.
4. Other 😄

Reply:
See the comment above.

Reply:
From now on, we only deploy to GitHub Pages after a merge to main or a manual CI run on main.